### PR TITLE
fix storageAt rpc call

### PR DESCRIPTION
**Pull request by yann300**
fix https://github.com/ethereum/remix-project/issues/2887

to test it:

 - deploy the Storage.sol
 - set it to a value
 - add a script in remix which uses `web3.eth.getStorage` end point (the slot should be 0)
 - the value should be the same as the value set.

### CODE REVIEW

File: libs/remix-simulator/src/VmProxy.ts [hunk 1]
```diff
@@ -294,19 +305,31 @@ export class VmProxy {
...
+    blockNumber = blockNumber === 'latest' ? this.vmContext.latestBlockNumber : blockNumber
 
-    let txHash
-    if (txIndex === 'latest') {
-      txHash = this.lastProcessedStorageTxHash[address]
-    } else {
-      const block = this.vmContext.blocks[blockNumber]
-      txHash = '0x' + block.transactions[txIndex].hash().toString('hex')
+    const block = this.vmContext.blocks[blockNumber]
+    const txHash = '0x' + block.transactions[block.transactions.length - 1].hash().toString('hex')
```
[the remark this answers is not in the document]
Comment (yann300) [reply libs/remix-simulator/src/VmProxy.ts:315]: I want the state of the storage at the start of the block or at the  end of the previous block.

File: libs/remix-simulator/test/blocks.ts [hunk 2]
```diff
@@ -212,13 +212,13 @@ describe('blocks', () => {
       let storage = await web3.eth.getStorageAt(contractInstance.options.address, 0)
       assert.deepEqual(storage, '0x64')
 
-      await contractInstance.methods.set(200).send({ from: accounts[0], gas: 400000 })
+      await contractInstance.methods.set(200).send({ from: accounts[0].toLowerCase(), gas: 400000 })
```
[the remark this answers is not in the document]
Comment (yann300) [reply libs/remix-simulator/test/blocks.ts:215]: I just do the same as above

**Comment by Aniket-Engg** [comment]
I used this script:
```
import Web3 from 'web3'

(async () => {
    try {
        const web3 = new Web3(web3Provider)
        const storage = await web3.eth.getStorageAt("0xd9145CCE52D386f254917e481eB44e9943F39138", 0)
        console.log(storage)
    } catch (e) {
        console.log(e.message)
    }
  })()
```

Looks working fine to me.

**Review by Aniket-Engg — APPROVED** [review]

**Comment by yann300** [comment]
yes so in live, `getStorage` would still return 0x0 even though the slot 0 is already set.

**Pull request merged**